### DISCUSSION TITLE
Formula loading disabled by HOMEBREW_DISABLE_LOAD_FORMULA!

### DISCUSSION THREAD

**Discussion by mahrud** [Homebrew/discussions#4546] [Tap maintenance and Homebrew development] [answered]
### Output of `brew config`

```shell
==> brew config
HOMEBREW_VERSION: 4.0.20-2-ga380aed
ORIGIN: https://github.com/Homebrew/brew
HEAD: a380aed887f40cbe29953269798d510502bf177d
Last commit: 23 hours ago
Core tap origin: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 3c77258a60e96cf75d81ac1c9eddda1058ba3850
Core tap last commit: 22 minutes ago
Core tap branch: master
HOMEBREW_PREFIX: /usr/local
HOMEBREW_BOOTSNAP: set
HOMEBREW_CACHE: /Users/runner/Library/Caches/Homebrew
HOMEBREW_CASK_OPTS: ["--no-quarantine"]
HOMEBREW_CLEANUP_PERIODIC_FULL_DAYS: 3650
HOMEBREW_COLOR: set
HOMEBREW_CURL_PATH: /usr/bin/curl
HOMEBREW_DEVELOPER: set
HOMEBREW_FAIL_LOG_LINES: 150
HOMEBREW_GIT_EMAIL: 1589480+BrewTestBot@users.noreply.github.com
HOMEBREW_GIT_NAME: BrewTestBot
HOMEBREW_GIT_PATH: /usr/bin/git
HOMEBREW_LOGS: /Users/runner/work/homebrew-tap/homebrew-tap/logs
HOMEBREW_MAKE_JOBS: 4
HOMEBREW_NO_AUTO_UPDATE: set
HOMEBREW_NO_EMOJI: set
HOMEBREW_NO_INSTALL_FROM_API: set
HOMEBREW_NO_INSTALLED_DEPENDENTS_CHECK: set
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: quad-core 64-bit ivybridge
Clang: 14.0.0 build 1400
Git: 2.37.1 => /usr/bin/git
Curl: 7.87.0 => /usr/bin/curl
macOS: 13.3.1-x86_64
CLT: 14.3.0.0.1.1679647830
Xcode: 14.2 => /Applications/Xcode_14.2.app/Contents/Developer
```

### Output of `brew doctor`

```shell
==> brew doctor
All steps passed!
```

### Description of issue

The github test-bot action keeps failing during `Run brew test-bot --only-formulae --verbose` step with the following error:
```
Error: Formula loading disabled by HOMEBREW_DISABLE_LOAD_FORMULA!
Error: Process completed with exit code 1.
```
I can't find any information about this flag, where it is set, or how to disable it, and nothing in my commit seems relevant. In fact, this is happening in two different pull requests with different formulae. Here is one: https://github.com/Macaulay2/homebrew-tap/actions/runs/5134412962/jobs/9238396226

**Comment by carlocab** [comment]
Can you [rerun your workflows](https://docs.github.com/en/actions/managing-workflow-runs/re-running-workflows-and-jobs?tool=webui) with debug logging enabled?

**Reply by mahrud** [reply]
It just ran [here](https://github.com/Macaulay2/homebrew-tap/actions/runs/5134412962/jobs/9238816908).

**Reply by carlocab** [reply, marked as answer]
Thanks, that was helpful. I should have a fix for this soon. As a workaround for now, you can do something like:
```yaml
env:
  HOMEBREW_GITHUB_API_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Using `GITHUB_TOKEN` may not be necessary; a non-empty string may suffice.

**Comment by carlocab** [comment]
Should be fixed by Homebrew/brew#15500.

**Reply by carlocab** [reply]
This should be fixed now. Let me know if you're still having problems with this.

**Reply by mahrud** [reply]
Thank you! I noticed that I now occasionally get the following error, for instance [here](https://github.com/Macaulay2/homebrew-tap/actions/runs/5146647104/jobs/9265988462). Is this related?
```bash
Error: GitHub API Error: API rate limit exceeded for 208.83.5.75. (But here's the good news: Authenticated requests get a higher rate limit. Check out the documentation for more details.)
Try again in 29 minutes 18 seconds, or:
Create a GitHub personal access token:
  https://github.com/settings/tokens/new?scopes=gist,repo,workflow&description=Homebrew
echo 'export HOMEBREW_GITHUB_API_TOKEN=your_token_here' >> ~/.profile
```
I can run it again in debug mode once all steps finish or fail.

**Reply by carlocab** [reply]
Yes. A workaround is to actually add this to your workflow:
```yaml
env:
  HOMEBREW_GITHUB_API_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
An empty string won't work this time. Let me think of a solution that doesn't require this.

**Reply by carlocab** [reply]
Here's a fix: https://github.com/Homebrew/homebrew-test-bot/pull/937
